test/hack/resource/pkg/resourcetypes/instance.go

```diff
@@ -120,12 +121,20 @@ func (i *Instance) Get(ctx context.Context, clusterName string) (ids []string, e
 
 // Cleanup any old instances that were managed by Karpenter or were provisioned as part of testing
 func (i *Instance) Cleanup(ctx context.Context, ids []string) ([]string, error) {
-	if _, err := i.ec2Client.TerminateInstances(ctx, &ec2.TerminateInstancesInput{
-		InstanceIds: ids,
-	}); err != nil {
-		return nil, err
+	// The maximum number of EC2 instances which can be specified in a single API call
+	const maxIDCount = 1000
+	cleaned := make([]string, 0, len(ids))
+	errs := make([]error, 0, len(ids)/maxIDCount+1)
```

Comment:
nit: Just to make this logic a little easier to parse -- does it make sense to chunk the list first and then just take the length of that to make the error slice?

Reply:
Sure, seems fine to me. I'll make the update.